File: packages/jsii-pacmak/test/expected.java/java/org/jsii/tests/calculator/Sum.java
package org.jsii.tests.calculator;
/**
 * An operation that sums multiple values.
 */
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii$jsii_calc$.Sum")
public class Sum extends org.jsii.tests.calculator.composition.CompositeOperation {
    protected Sum(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public Sum() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    /**
     * The parts to sum.
     */
    public java.util.List<org.jsii.tests.calculator.lib.Value> getParts() {
        return this.jsiiGet("parts", java.util.List.class);
    }
    /**
     * The parts to sum.
     */
    public void setParts(final java.util.List<org.jsii.tests.calculator.lib.Value> value) {
        this.jsiiSet("parts", value);
    }
    /**
     * The expression that this operation consists of.
     * Must be implemented by derived classes.
     */
    public org.jsii.tests.calculator.lib.Value getExpression() {
        return this.jsiiGet("expression", org.jsii.tests.calculator.lib.Value.class);
    }
}

File: packages/jsii-pacmak/test/expected.dotnet/AWS.Jsii.Tests.Calculator/Sum.cs
using AWS.Jsii.Runtime.Deputy;
using AWS.Jsii.Tests.Calculator.composition;
using AWS.Jsii.Tests.Calculator.Lib;

namespace AWS.Jsii.Tests.Calculator
{
    /// <summary>An operation that sums multiple values.</summary>
    [JsiiClass("jsii-calc", "jsii$jsii_calc$.Sum", "[]")]
    public class Sum : CompositeOperation
    {
        public Sum(): base(new DeputyProps(new object[]{}))
        {
        }

        protected Sum(ByRefValue reference): base(reference)
        {
        }

        protected Sum(DeputyProps props): base(props)
        {
        }

        /// <summary>The parts to sum.</summary>
        [JsiiProperty("parts", "{\"collection\":{\"kind\":\"array\",\"elementtype\":{\"fqn\":\"jsii$jsii_calc_lib$.Value\"}}}")]
        public virtual Value_[] Parts
        {
            get => GetProperty<Value_[]>();
            set => SetProperty(value);
        }

        /// <summary>
        /// The expression that this operation consists of.
        /// Must be implemented by derived classes.
        /// </summary>
        [JsiiProperty("expression", "{\"fqn\":\"jsii$jsii_calc_lib$.Value\"}")]
        public override Value_ Expression
        {
            get => GetProperty<Value_>();
        }
    }
}